src/build_packages.sh

```diff
@@ -2,27 +2,69 @@
 set -e
 export XROAD=$(cd "$(dirname "$0")"; pwd)
 
-if [[ ! " $* " =~ " --packageonly " ]]; then
-  ./compile_code.sh "$@"
-fi
+HAS_DOCKER=""
+
+errorExit() {
+    echo "*** $*" 1>&2
+    exit 1
+}
+
+usage () {
+    echo "Usage: $0 [option for $0...] [other options]"
+    echo "Options for $0:"
+    echo " -p, --package-only    Skip compilation, just build packages'"
+    echo " -d, --docker-compile  Compile in docker container instead of native gradle build"
+    echo " -h, --help            This help text."
+    echo "The option for $0, if present, must come fist, before other options."
+    echo "Other options are passed on to compile_code.sh"
+    test -z "$1" || exit "$1"
+}
+
+buildInDocker() {
+    test -n "$HAS_DOCKER" || errorExit "Error, docker is not installed/running."
+    echo "Building in docker..."
+    # check if running attached to terminal
+    # makes it possible to stop build with Ctrl+C
+    if [ -t 1 ]; then OPT="-it"; fi
+
+    docker build -q -t xroad-build --build-arg uid=$(id -u) --build-arg gid=$(id -g) $XROAD/packages/docker-compile || errorExit "Error building build image."
+    docker run --rm -v $XROAD/..:/workspace -w /workspace/src -u builder ${OPT} xroad-build  bash -c "./update_ruby_dependencies.sh && ./compile_code.sh -nodaemon" || errorExit "Error running build of binaries."
+}
+
+buildLocally() {
+    echo "Building locally..."
+    cd $XROAD || errorExit "Error 'cd $XROAD'."
+    ./compile_code.sh "$@" || errorExit "Error running build of binaries."
+}
 
 if command -v docker &>/dev/null; then
-    docker build -q -t xroad-deb-bionic "$XROAD/packages/docker/deb-bionic"
-    docker build -q -t xroad-deb-focal "$XROAD/packages/docker/deb-focal"
-    docker build -q -t xroad-rpm "$XROAD/packages/docker/rpm"
-    docker build -q -t xroad-rpm-el8 "$XROAD/packages/docker/rpm-el8"
+    HAS_DOCKER=true
+fi
+
+case "$1" in
+    --packageonly|-p) shift;;
```

Comment:
Actual expected parameters `packageonly` and `docker-build` do not match now what usage says (`package-only` and `docker-compile`).

I would use the names from usage. Change from packageonly to package-only is in principle a breaking change, but has been an undocumented feature and by a quick check should not cause trouble for our CI jobs, at least.

Reply:
Sure, that had slipped my attention.